ICS2: Make Synopsis and Motivation clearer (#713)
cosmos/ibc: spec/core/ics-002-client-semantics/README.md

## Synopsis

This standard specifies the properties that consensus algorithms of state machines implementing the inter-blockchain
communication (IBC) protocol are required to satisfy. 
These properties are necessary for efficient and safe verification in the higher-level protocol abstractions. 

> The algorithm utilised in IBC to verify the state updates of a remote state machine is referred to as a *validity predicate*. 

Reviewer: maybe we should consider changing this term to "state predicate", given that we use "misbehaviour predicate" below? "validity predicate" in this context is a bit redundant

Author: There are two predicates: a validity predicate and a misbehaviour predicate. The validity predicate is used to check the **_validity_** of incoming headers, which are then used to update the ConsensusState. Thus, I think the name is good. A "state predicate" would entail that the predicate is used to checks the state of the remote chain.

Pairing a validity predicate with a trusted state (i.e., a state that the verifier assumes to be correct), 
forms a *light client*, often shortened to *client*.
In addition to state update verification, every light client is able to detect consensus misbehaviours through a *misbehaviour predicate*.

Beyond the properties described in this specification, IBC does not impose any requirements on
the internal operation of the state machines and their consensus algorithms. 
A state machine may consist of a single process signing operations with a private key, a quorum of processes signing in unison,

> many processes operating a Byzantine fault-tolerant consensus algorithm, or other configurations yet to be invented

Reviewer 2: ```suggestion
many processes operating a Byzantine fault-tolerant consensus algorithm (e.g., Tendermint), or other configurations yet to be invented
```

— from the perspective of IBC, a state machine is defined entirely by its light client validation and misbehaviour detection logic.

> This standard also specifies how light clients are stored, registered, and updated by the IBC protocol. 

Reviewer 2: ```suggestion
This standard also specifies how the light client's functionality is registered and how it's data is stored and updated by the IBC protocol. 
```

* `ConsensusState` is an opaque type representing the state of a validity predicate.
  `ConsensusState` must be able to verify state updates agreed upon by the associated consensus algorithm.

>   It must also be serialisable in a canonical fashion so that third parties, such as counterparty state machines,

Reviewer 2: for counterparty the term remote was used before. Perhaps we should fix one, and use it everywhere?

Author: ```suggestion
  It must also be serialisable in a canonical fashion so that third parties, such as remote state machines,
```
